cloud/fabfile.py

```diff
@@ -46,9 +46,29 @@ def proxy():
     env.ssh_config_path = "../client/ssh/proxy-config"
     env.use_ssh_config = True
 
+def is_hg():
+    #first test for hg
+    try:
+        local("hg identify")
+        return True
+    except:
+        return False
+
+def is_git():
+    #first test for hg
+    try:
+        local("git rev-parse")
+        return True
+    except:
+        return False
 
 def get_bookmark():
```

Comment:
I'd make a version of this function that gets all the metadata used in the `package()` function - bookmark/branch (feel free to also make the terminology consistent throughout), version, commit summary - and returns it as a tuple or dict, so that all the git vs hg stuff can be kept in one place. Then you don't need the new `is_foo` functions. `get_bookmark()` would just call that new function and return the bookmark only, for compatibility.

Note that I think function names in the top-level fabfile can (should?) be prefixed with an underscore if they are for internal use only and don't need to be made available as commands. `get_bookmark`/`get_metadata` probably fit into that category, unless they can usefully be run directly for debugging (but whether that justifies keeping them 'exported' is debatable); `is_foo` should be made internal only (if you aren't going to get rid of it/them).